src/lib.rs

```diff
@@ -408,61 +408,155 @@
-//! command description inside the parents help message and the long version
-//! as command description if help is requested on the actual subcommand.
-//!
-//! This feature can also be used with doc comments instead of arguments through
-//! proper comment formatting. To be activated it requires, that the first line
-//! of the comment is separated from the rest of the comment through an empty line.
-//! In this case the first line is used as summary and the whole comment represents
-//! the long descriptive message.
+//! For convenience, doc comments can be used instead of raw methods
+//! (this example works exactly as one above):
 //!
 //! ```
 //! # use structopt::StructOpt;
 //!
 //! #[derive(StructOpt)]
-//! /// The help message that will be displayed when passing `--help`.
+//! /// I am a program and I work, just pass `-h`
 //! struct Foo {
-//!   #[structopt(short)]
-//!   /// Only this summary is visible when passing `-h`.
-//!   ///
-//!   /// But the whole comment will be displayed when passing `--help`.
-//!   /// This could be quite useful to provide further hints are usage
-//!   /// examples.
+//!   /// Pass `-h` and you'll see me!
 //!   bar: String
 //! }
-//! # fn main() {}
 //! ```
 //!
+//! Doc comments on [top-level](#magical-methods) will be turned into
+//! `App::about/long_about` call (see below), doc comments on field-level are
+//! `Arg::help/long_help` calls.
+//!
+//! **Note:**
+//! _________________
+//!
+//! Raw methods have priority over doc comments!
+//!
+//! **Top level doc comments always generate `App::about/long_about` calls!**
+//! If you really want to use `App::help/long_help` method (you likely don't) use
+//! a raw method with no doc comment.
+//! __________________
+//!
+//! ### `long_help` and `--help`
+//!
+//! A message passed to [`Arg::long_help`]/[`App::long_about`] get displayed whenever
+//! your program is called with `--help` instead of `-h`. Of course, you can
+//! use them via raw methods as described [above](#help-messages).
+//!
+//! The more convenient way is to use so-called "long" doc comment:
+//!
+//! ```
+//! # use structopt::StructOpt;
+//! #[derive(StructOpt)]
+//! /// Hi there, I'm Robo!
+//! ///
+//! /// I like beeping, stumbling, eating your electricity,
+//! /// and making records of you singing in a shower.
+//! /// Pay up or I'll upload it to youtube!
+//! struct Robo {
+//!     /// Call my brother SkyNet.
+//!     ///
+//!     /// I am artificial superintelligence. I won't rest
+//!     /// until I'll have destroyed humanity. Enjoy your
+//!     /// pathetic existence, you mere mortal.
+//!     #[structopt(long)]
+//!     kill_all_humans: String
+//! }
+//! ```
+//!
+//! A long doc comment consists of three parts:
+//! * Short summary
+//! * A blank line (can contain only whitespace)
+//! * Detailed description, all the rest
+//!
+//! I.e "long" doc comment consists of two paragraphs, first is summary,
```

Comment:
> Don't use `i.e.` outside of parentheses and never capitalize it.

Good to know, thanks!

> I'm assuming you don't also have an artificial restriction on how many paragraphs the detailed description may be.

There's no such restriction, but current implementation does not preserve such empty lines, collapsing the whole rest into one giant abstract. I'm fixing it, thanks for pointing out.

And a question: if two paragraphs are delimited by *more than one blank line in a row* what do you think we should do? Preserve them all or collapse into one blank line?

Reply:
Collapse them into one blank line and document that.

When in doubt, I recommend following how HTML renderers handle whitespace in raw HTML.

Reply:
I hate articles, that's official.

Reply:
\*chuckle* I can sort of understand how you feel, because I'm learning French, which is even less forgiving of omitting them. (Plenty of places in English were you can use a noun and French needs you to include either the definite (le/la/les) or indefinite (de/des) article.

(eg. In English, you can say "I want spaghetti". In French, you have to be specific and say "I want **some** spaghetti", not "I want **the** spaghetti". If nothing else, it really gets you thinking about what you actually *mean* in English.)

Reply:
> I can sort of understand how you feel, because I'm learning French, which is even less forgiving of omitting them.

Honestly, I'd be fine with using articles with every single noun in a goddamn sentence, but there is a number of rules (*of course*) when I should omit them... Why the hell do you even need them? Isn't it obvious that "restaurants" is a countable plural noun? 

When I'll have conquered the world I'll ditch them from English, I promise (got lost in dreams...).



>  not "I want **the** spaghetti"

But want if you want *this specific portion on the table just in front of you*?

Hey @TeXitoi , would you kindly explain why the hell French people need articles?

Reply:
> Honestly, I'd be fine with using articles with every single noun in a goddamn sentence, but there is a number of rules (of course) when I should omit them... 

I haven't stopped to look for counter-examples but my first impulse is to say "Always including the implied article is fine... as long as you don't mind sounding conspiciously formal".

> Why the hell do you even need them?

When the rest of the sentence is providing adequate context? To avoid sounding like Hollywood's impression of a cave man.

When they're actually useful, to emphasize subtle distinctions (similar to how, in the past, "Do I dare?" was a version of "Dare I?" that placed emphasis on the "Do") or to allow other parts of the sentence to be made more concise without it becoming unclear. (eg. When talking about uncountable nouns like bread or water, "a/an" tends to be used to refer to abstract concepts, as in "a cheese" or "a bread" being shorthand for "a type of cheese" (eg. brie) and "a type of bread" (eg. ciabatta) or to draw attention to wordplay.)

> Isn't it obvious that "restaurants" is a countable common plural noun?

Funny that you'd pick a plural noun.

In the most common cases, making something plural removes the need to use an explicit article. (That's why we think "Japanese doesn't use articles or plurals, so Neko means 'a cat', 'the cat', or 'cats' depending on context", rather than "Japanese doesn't use articles or plurals, so Neko means 'a cat', 'the cat', or, 'some cats', or 'the cats' depending on context".)

Anyway, give it a few centuries and that might change. Given that it's on its way to being the international lingua franca, it's already simplifying. (eg. throwing out the distinction between "who" and "whom" outside formal speech in favour of using "who" everywhere, because it adds no information that can't be gleaned from the sentence structure, or discarding the masculine/feminine distinction between "fiance" and "fiancee" in favour of the feminine form which makes the French-derived pronunciation clear without needing to write it with an accent, as in "fiancé" and "fiancée"... a recurring pattern in French-to-English loanwords, by the way. See also actif/active, for example.)

> But what if you want this specific portion on the table just in front of you?

Then you'd use "the" in English too. My point is that learning French made me realize that, in that English sentence, there's an implicit "some".